static/app/utils/formatters.tsx

```diff
@@ -301,12 +310,21 @@ export function formatAbbreviatedNumber(number: number | string) {
       continue;
     }
 
-    return shortValue / 10 > 1 || !fitsBound
-      ? `${shortValue}${suffix}`
-      : `${formatFloat(number / suffixNum, 1)}${suffix}`;
+    const formattedNumber =
+      shortValue / 10 > 1 || !fitsBound
+        ? shortValue
+        : formatFloat(number / suffixNum, precision || 1);
+
+    const formattedNumberWithPercision =
+      precision === undefined
+        ? formattedNumber
+        : parseFloat(formattedNumber.toPrecision(precision));
```

Comment:
I suggested `toPrecision` in my comment but I was straight-up wrong, I think. The right move is actually this:
```suggestion
        : parseFloat(formattedNumber.toLocaleString(undefined, {maximumSignificantDigits: 3. minimumSignificantDigits: 3}));
```

This will have the right number of significant digits _and_ it'll use correct locale formatting 👌🏻 Same for the bit below, too

Reply:
TIL about this, will change!

Reply:
Should I add `toLocaleString` even when not using precision? This will lead to numbers like `1,200b` being generated by `formatAbbreviatedNumber`.

I chose not to in f40947fcbfce1605b9238492f4f325b0d1d1e9d3

Reply:
I think choosing not to is totally fine. I'm going to bring up number formatting with the FE TSC today, too, to see how everyone usually tackles this